src/vs/workbench/api/common/extHostTelemetry.ts

```diff
@@ -8,38 +8,53 @@ import { createDecorator } from 'vs/platform/instantiation/common/instantiation'
+import { cleanData, cleanRemoteAuthority, extensionTelemetryLogChannelId } from 'vs/platform/telemetry/common/telemetryUtils';
 import { mixin } from 'vs/base/common/objects';
 import { URI } from 'vs/base/common/uri';
+import { Disposable } from 'vs/base/common/lifecycle';
+import { localize } from 'vs/nls';
 
-export class ExtHostTelemetry implements ExtHostTelemetryShape {
-	private readonly _onDidChangeTelemetryEnabled = new Emitter<boolean>();
+export class ExtHostTelemetry extends Disposable implements ExtHostTelemetryShape {
+	private readonly _onDidChangeTelemetryEnabled = this._register(new Emitter<boolean>());
 	readonly onDidChangeTelemetryEnabled: Event<boolean> = this._onDidChangeTelemetryEnabled.event;
 
-	private readonly _onDidChangeTelemetryConfiguration = new Emitter<vscode.TelemetryConfiguration>();
+	private readonly _onDidChangeTelemetryConfiguration = this._register(new Emitter<vscode.TelemetryConfiguration>());
 	readonly onDidChangeTelemetryConfiguration: Event<vscode.TelemetryConfiguration> = this._onDidChangeTelemetryConfiguration.event;
 
 	private _productConfig: { usage: boolean; error: boolean } = { usage: true, error: true };
 	private _level: TelemetryLevel = TelemetryLevel.NONE;
 	private _oldTelemetryEnablement: boolean | undefined;
-	private _inLoggingOnlyMode: boolean = false;
+	private readonly _inLoggingOnlyMode: boolean = false;
+	private readonly extHostTelemetryLogFile: URI;
 	private readonly _outputLogger: ILogger;
 	private readonly _telemetryLoggers = new Map<string, ExtHostTelemetryLogger>();
 
 	constructor(
 		@IExtHostInitDataService private readonly initData: IExtHostInitDataService,
-		@ILoggerService loggerService: ILoggerService,
+		@ILoggerService private readonly loggerService: ILoggerService,
 	) {
-		this._outputLogger = loggerService.createLogger(URI.revive(this.initData.environment.extensionTelemetryLogResource), { hidden: true });
+		super();
+		this.extHostTelemetryLogFile = URI.revive(this.initData.environment.extensionTelemetryLogResource);
+		this._inLoggingOnlyMode = this.initData.environment.isExtensionTelemetryLoggingOnly;
+		this._outputLogger = loggerService.createLogger(this.extHostTelemetryLogFile, { id: extensionTelemetryLogChannelId, name: localize('extensionTelemetryLog', "Extension Telemetry{0}", this._inLoggingOnlyMode ? ' (Not Sent)' : ''), hidden: true });
+		this._register(loggerService.onDidChangeLogLevel(arg => {
+			if (isLogLevel(arg)) {
+				this.updateLoggerVisibility();
+			}
+		}));
 		this._outputLogger.info('Below are logs for extension telemetry events sent to the telemetry output channel API once the log level is set to trace.');
 		this._outputLogger.info('===========================================================');
 	}
 
+	private updateLoggerVisibility(): void {
+		this.loggerService.setVisibility(this.extHostTelemetryLogFile, this._level !== TelemetryLevel.NONE && this.loggerService.getLogLevel() === LogLevel.Trace);
```

Comment:
We would want to show the log even when telemetry level is none because we want to show the users that nothing is flowing. We'd want to use the same `supportsTelemetry` function as before  as that's checking for `--disable-telemetry` and things of that nature.